lib/logqueue-fifo.c

```diff
@@ -489,6 +489,7 @@ log_queue_fifo_pop_head(LogQueue *s, LogPathOptions *path_options)
   if (!node->flow_control_requested)
     self->backlog_queue.non_flow_controlled_len++;
 
+  msg_set_context(msg);
```

Comment:
Why don't you place this in the log queue abstract pop method? That way, the memory queues and disk buffers won't have to be careful about this.